### ISSUE TITLE
e2e flake: deployment should label adopted RSs and pods [pods not labeled]

### ISSUE THREAD

**Issue by bgrant0607** [kubernetes/kubernetes#22302]
kubernetes-jenkins/logs/kubernetes-e2e-gce-ubernetes-lite/639

```
Deployment
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/deployment.go:72
  deployment should label adopted RSs and pods [It]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/deployment.go:71

  Expected
      <int>: 0
  to be >
      <int>: 0

  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/deployment.go:878
```

The error message is confusing, but that's expected to be > 0:
```
	newRS, err := deploymentutil.GetNewReplicaSet(deployment, c)
	Expect(err).NotTo(HaveOccurred())
	Expect(len(newRS.Labels[extensions.DefaultDeploymentUniqueLabelKey])).Should(BeNumerically(">", 0))
	Expect(len(newRS.Spec.Selector.MatchLabels[extensions.DefaultDeploymentUniqueLabelKey])).Should(BeNumerically(">", 0))
	Expect(len(newRS.Spec.Template.Labels[extensions.DefaultDeploymentUniqueLabelKey])).Should(BeNumerically(">", 0))
	// All pods targeted by the deployment should contain pod-template-hash in their labels, and there should be only 3 pods
	selector, err := unversioned.LabelSelectorAsSelector(deployment.Spec.Selector)
	Expect(err).NotTo(HaveOccurred())
	options := api.ListOptions{LabelSelector: selector}
	pods, err := c.Core().Pods(ns).List(options)
	Expect(err).NotTo(HaveOccurred())
	for _, pod := range pods.Items {
		Expect(len(pod.Labels[extensions.DefaultDeploymentUniqueLabelKey])).Should(BeNumerically(">", 0))
	}
	Expect(len(pods.Items)).Should(Equal(replicas))
```
In the case of failure, it would be useful to print the pod that failed.

cc @janetkuo @nikhiljindal @kargakis

**Comment by mml** [comment]
I'd like to work on this, but as a n00b I may make slow progress.  If anyone cares to take this from me I won't be offended.

**Comment by janetkuo** [comment]
> In the case of failure, it would be useful to print the pod that failed.

Filed #22312 for logging it

**Comment by bgrant0607** [comment]
#22300 shows another case where a pod doesn't have the pod-template-hash label added.

**Comment by nikhiljindal** [comment]
https://github.com/kubernetes/kubernetes/pull/22305 fixed a bug due to which template hash was not being added to pods.

**Comment by janetkuo** [comment]
This should be fixed by #22305. Please reopen if it happens again.
